components/cam/src/physics/cam/froude.F90

```diff
@@ -0,0 +1,176 @@
+    use cam_history,    only: outfld
+    use wv_saturation,  only: qsat
+    use camsrfexch,     only: cam_out_t
+!
+!     i/o fields
+!
+    type(physics_state), intent(in) :: state
+    type (cam_out_t), intent(inout) :: cam_out
+
+!     local fields
+
+    real(r8) :: uovern(pcols)  ! ratio of wind speed/brunt vaisalla frequency
+    real(r8) :: thx(pverp) ! potential temperature
+    real(r8) :: t(pver) ! temperature
+    real(r8) :: tint(pverp) ! temperature at layer interfaces
+    real(r8) :: q(pver) ! specific humidity
+    real(r8) :: qint(pverp) ! specific humidity at layer interfaces
+    real(r8) :: bvf(pver) ! brunt-vaisala frequency
+    real(r8) :: pmid(pver) ! pressure at model levels
+    real(r8) :: pint(pverp)   ! Interface pressures
+    real(r8) :: lnpint(pverp) ! ln(pressure) at model interfaces
+    real(r8) :: pdel(pver) ! pdel(k)   = pint  (k+1)-pint  (k)
+    real(r8) :: zs ! surface elevation
+    real(r8) :: zint(pverp) ! height at model interfaces
+    real(r8) :: zmid(pver)  ! height at model levels
+    real(r8) :: bvfmin ! minimum brunt-vaisala frequency
+    real(r8) :: wspeed ! wind speed
+    real(r8) :: cmass ! column mass
+    real(r8) :: wspdbar, bvfbar
+    real(r8) :: gam, gammadry, gammawet
+    real(r8) :: es,qs,rh(pver)
+    integer  :: k, kmax, kmin  ! layer index
+    integer  :: i, n,m  ! constituent index
+    real(r8) :: rovg  ! rair/gravit
+    real(r8) :: d
+    integer :: ncol, lchnk
+
+    character(len=*), parameter      :: subname = "froude"
+
+   ncol  = state%ncol
+
+
+   do i=1,ncol
+          do k = 1, pverp
+            pint(k) = hyai(k)*ps0 + hybi(k)*state%ps(i)
+            lnpint(k) = log(pint(k))
+          end do
+          do k = 1, pver
+            pdel(k) = pint(k+1) - pint(k)
+            pmid(k) = 0.5_r8*(pint(k) + pint(k+1))
+            t(k)=state%t(i,k)
+            q(k)=state%q(i,k,1)
+          end do
+          kmax=pver-1
+          do k=kmax-1,1,-1
+             if(pint(kmax)-pint(k).gt.2.e4)then
+                kmin=k
+                exit
+             end if 
+          end do
+
+          do k = 1, pver
+            call qsat(t(k),pmid(k),es,qs)
+            rh(k)=q(k)/qs
+          end do
+!
+!---- determine the brunt-vaisala frequency
+!
+          do k = 2, pver
+            thx(k)=0.5*(state%t(i,k)*(state%ps(i)/pmid(k))**cappa+state%t(i,k-1)*(state%ps(i)/pmid(k-1))**cappa)
+            tint(k)=0.5*(state%t(i,k)+state%t(i,k-1))
+            qint(k)=0.5*(state%q(i,k,1)+state%q(i,k-1,1))
+          end do
+!
+          thx(pver+1) = state%t(i,pver)
+          tint(pver+1) = state%t(i,pver)
+          qint(pver+1) = state%q(i,pver,1)
+!
+          zs = state%phis(i) / gravit
+          zint(pver+1) = zs
+!
+          rovg=rair/gravit
+          do k = pver, 1, -1
+            zint(k) = zint(k+1)+                             &
+                 rovg*state%t(i,k)*(1. + zvir * state%q(i,k,1))*(lnpint(k+1)-lnpint(k))
```

Comment:
Same for 1. . It should be 1.0_r8

Reply:
Thanks! Requested changes have been applied.